content/docs/rfcs/12/README.md

```diff
@@ -61,77 +63,148 @@ The following are not considered as part of the adversarial model:
 ## Protobuf
 
 ```protobuf
-message FilterRequest {
-  bool subscribe = 1;
-  string topic = 2;
-  repeated ContentFilter contentFilters = 3;
-
-  message ContentFilter {
-    string contentTopic = 1;
+syntax = "proto3";
+
+// 12/WAKU2-FILTER rfc: https://rfc.vac.dev/spec/12/
+package waku.filter.v2;
+
+// Protocol identifier: /vac/waku/filter-subscribe/2.0.0-beta1
+message FilterSubscribeRequest {
+  enum FilterSubscribeType {
+    SUBSCRIBER_PING = 0;
+    SUBSCRIBE = 1;
+    UNSUBSCRIBE = 2;
+    UNSUBSCRIBE_ALL = 3;
   }
+
+  string request_id = 1;
+  FilterSubscribeType filter_subscribe_type = 2;
+
+  // Filter criteria
+  optional string pubsub_topic = 10;
+  repeated string content_topics = 11;
 }
 
-message MessagePush {
-  repeated WakuMessage messages = 1;
+message FilterSubscribeResponse {
+  string request_id = 1;
+  uint32 status_code = 10;
+  optional string status_desc = 11;
 }
 
-message FilterRPC {
-  string requestId = 1;
-  FilterRequest request = 2;
-  MessagePush push = 3;
+// Protocol identifier: /vac/waku/filter-push/2.0.0-beta1
+message MessagePush {
+  WakuMessage waku_message = 1;
```

Comment:
```suggestion
  repeated WakuMessage waku_message = 1;
```

Reply:
The previous protobuf indeed had several messages per push. However, since filter service node does not cache messages they should forward messages as they arrive (similar to just routing them on relay). This is also how it's currently implemented - each MessagePush only ever contains a single WakuMessage, even if it's a repeated field. WDYT?

Reply:
Ah! I agree then. My previous comment can be dismissed then. Thank you!